[BottomSheet] Example unit tests for updatePreferredSheetHeight - WIP

romoore opened the pull request:
A couple of basic unit tests to verify the logic flows inside `updatePreferredSheetHeight`. They could definitely benefit from some refactoring within MDCBottomSheetPresentationController, but given the current code structure these seem to be the simplest way to exercise the new code independently.

Marked "WIP" so nobody tries to merge this as it's just an example to illustrate my thoughts on #5023

codeman7 committed:
Intial commit

codeman7 committed:
Address feedback

codeman7 committed:
Add test

codeman7 committed:
Work on test

Commit:
[BottomSheet] Tests for updatePreferredSheetHeight

Adding 2 basic tests for `updatePreferredSheetHeight` to verify that the
method uses the various view/viewcontroller properties as expected.

1. When `presentedViewController` has a `preferredContentSize`
2. When (1) is false, but the `sheetView` has a non-zero bounds

material-automation commented on components/BottomSheet/tests/unit/MDCBottomSheetPresentationControllerTests.m:
```diff
@@ -0,0 +1,124 @@
+  // Given
+  FakeSheetView *sheetView = [[FakeSheetView alloc] init];
+  UIViewController *presentingViewController = [[UIViewController alloc] init];
+  MDCFakePresentedViewController *presentedViewController = [[MDCFakePresentedViewController alloc] init];
```
clang-format suggested the following change:


```
-  MDCFakePresentedViewController *presentedViewController = [[MDCFakePresentedViewController alloc] init];
+  MDCFakePresentedViewController *presentedViewController =
+      [[MDCFakePresentedViewController alloc] init];
```

<!-- identifier: clang-format -->
<!-- File: components/BottomSheet/tests/unit/MDCBottomSheetPresentationControllerTests.m -->

material-automation commented on components/BottomSheet/tests/unit/MDCBottomSheetPresentationControllerTests.m:
```diff
@@ -0,0 +1,124 @@
+  MDCFakeBottomSheetPresentationController *presentationController =
+      [[MDCFakeBottomSheetPresentationController alloc]
+          initWithPresentedViewController:presentedViewController
+          presentingViewController:presentingViewController];
```
clang-format suggested the following change:


```
-          presentingViewController:presentingViewController];
+                 presentingViewController:presentingViewController];
```

<!-- identifier: clang-format -->
<!-- File: components/BottomSheet/tests/unit/MDCBottomSheetPresentationControllerTests.m -->

material-automation commented on components/BottomSheet/tests/unit/MDCBottomSheetPresentationControllerTests.m:
```diff
@@ -0,0 +1,124 @@
+  // Given
+  FakeSheetView *sheetView = [[FakeSheetView alloc] init];
+  UIViewController *presentingViewController = [[UIViewController alloc] init];
+  MDCFakePresentedViewController *presentedViewController = [[MDCFakePresentedViewController alloc] init];
```
clang-format suggested the following change:


```
-  MDCFakePresentedViewController *presentedViewController = [[MDCFakePresentedViewController alloc] init];
+  MDCFakePresentedViewController *presentedViewController =
+      [[MDCFakePresentedViewController alloc] init];
```

<!-- identifier: clang-format -->
<!-- File: components/BottomSheet/tests/unit/MDCBottomSheetPresentationControllerTests.m -->

material-automation commented on components/BottomSheet/tests/unit/MDCBottomSheetPresentationControllerTests.m:
```diff
@@ -0,0 +1,124 @@
+  MDCFakeBottomSheetPresentationController *presentationController =
+  [[MDCFakeBottomSheetPresentationController alloc]
+   initWithPresentedViewController:presentedViewController
+   presentingViewController:presentingViewController];
```
clang-format suggested the following change:


```
-  [[MDCFakeBottomSheetPresentationController alloc]
-   initWithPresentedViewController:presentedViewController
-   presentingViewController:presentingViewController];
+      [[MDCFakeBottomSheetPresentationController alloc]
+          initWithPresentedViewController:presentedViewController
+                 presentingViewController:presentingViewController];
```

<!-- identifier: clang-format -->
<!-- File: components/BottomSheet/tests/unit/MDCBottomSheetPresentationControllerTests.m -->

material-automation commented:
clang-format recommended changes.

Consider installing [git-clang-format](https://github.com/llvm-mirror/clang/blob/master/tools/clang-format/git-clang-format) and running the following command:

```bash
git clang-format $(git merge-base origin/develop HEAD)
```

<!-- identifier: clang-format -->

romoore commented on components/BottomSheet/tests/unit/MDCBottomSheetPresentationControllerTests.m:
```diff
@@ -0,0 +1,124 @@
+      [[MDCFakeBottomSheetPresentationController alloc]
+          initWithPresentedViewController:presentedViewController
+          presentingViewController:presentingViewController];
+  presentationController.test_PresentedViewController = presentedViewController;
```
This extra assignment/property may not even be needed given that the initializer probably just assigns the property.

romoore replied:
Yeah, it can.

Commit:
Add another test.

Adding a test for when an explicit preferredSheetHeight is set on the
presentation controller. Also removed an unnecessary faked property for
`presentedViewController`.

codeman7 commented:
I don't see why not to merge. It adds the functionality and test that we need.

codeman7 commented on components/BottomSheet/tests/unit/BottomSheetPresentationTest.swift:
```diff
@@ -0,0 +1,69 @@
+    presentationController.preferredSheetHeight = preferredSheetHeight
+    presentationController.presentedViewController.view.setNeedsLayout()
+    presentationController.presentedViewController.view.layoutIfNeeded()
+    print(presentationController.frameOfPresentedViewInContainerView)
```
Please remove before merge.

romoore commented:
Made obsolete by #5115
